draft-ietf-quic-qpack.md

```diff
@@ -52,46 +52,162 @@ code and issues list for this draft can be found at
+entries are defined in Appendix A of {{!RFC7541}}. Note that because HPACK did
+not use zero-based references, there is no value at index zero of the static
+table.
+
+### Dynamic Table {#table-dynamic}
+
+The dynamic table consists of a list of header fields maintained in first-in,
+first-out order.  The dynamic table is initially empty.  Entries are added by
+instructions on the Encoder Stream (see {{encoder-stream}}).
+
+Before a new entry is added to the dynamic table, entries are evicted from the
+end of the dynamic table until the size of the dynamic table is less than or
+equal to (maximum size - new entry size) or until the table is empty.
+
+If the size of the new entry is less than or equal to the maximum size, that
+entry is added to the table.  It is not an error to attempt to add an entry that
+is larger than the maximum size; an attempt to add an entry larger than the
+maximum size causes the table to be emptied of all existing entries and results
+in an empty table.
+
+A new entry can reference the name of an entry in the dynamic table that will be
+evicted when adding this new entry into the dynamic table.  Implementations are
+cautioned to avoid deleting the referenced name if the referenced entry is
+evicted from the dynamic table prior to inserting the new entry.
+
+The dynamic table can contain duplicate entries (i.e., entries with the same
+name and same value).  Therefore, duplicate entries MUST NOT be treated as an
+error by a decoder.
+
+The encoder decides how to update the dynamic table and as such can control how
+much memory is used by the dynamic table.  To limit the memory requirements of
+the decoder, the dynamic table size is strictly bounded.
+
+The decoder determines the maximum size that the encoder is permitted to use for
+the dynamic table.  In HTTP/QUIC, this value is determined by the
+SETTINGS_HEADER_TABLE_SIZE setting (see Section 4.2.5.2 of {{!QUIC-HTTP}}).
+
+An encoder can choose to use less capacity than this maximum size (see
+{{size-update}}), but the chosen size MUST stay lower than or equal to the
+maximum set by the decoder.  Whenever the maximum size for the dynamic table is
+reduced, entries are evicted from the end of the dynamic table until the size of
+the dynamic table is less than or equal to the maximum size.
+
+This mechanism can be used to completely clear entries from the dynamic table by
+setting a maximum size of 0, which can subsequently be restored.
+
+### Indexing
+
+Entries in the QPACK static and dynamic tables are addressed separately.
+
+While entries in the static table have the same indices at all times, entries
+are inserted into the dynamic table over time.  Each entry possesses both an
+absolute index which is fixed for the lifetime of that entry and a relative
+index which changes over time based on the context of the reference. The first
+entry inserted has an absolute index of "1"; indices increase sequentially with
+each insertion.
```

Comment:
You need to explain that relative indexes increase in the opposite direction to the absolute index.  This is the most confusing aspect of this entire thing.